src/mappings/utils.ts

```diff
@@ -130,12 +134,53 @@ export async function trackUnprocessed(error: Error, primitives: Primitives): Pr
   }
 }
 
+// Returns the id of the entity that establishes the relationship between the MsgStake and Service.
+export function getMsgStakeServiceId(msgStakeId: string, serviceId: string): string {
+  return `${msgStakeId}-${serviceId}`;
+}
+
+// Returns the id of the entity that establishes the relationship between the Entity staked and Service.
+export function getStakeServiceId(entityStakedId: string, serviceId: string): string {
+  return `${entityStakedId}-${serviceId}`;
+}
+
 // getBalanceId returns the id of the Balance entity using the address and denom passed.
 // Use this to get the id of the Balance entities across the indexing process.
 export function getBalanceId(address: string, denom: string): string {
   return `${address}-${denom}`;
 }
 
+// Returns the id of the entity that establishes the relationship between the Gateway that the app is being delegated to.
+export function getAppDelegatedToGatewayId(appAddress: string, gatewayAddress: string): string {
+  return `${appAddress}-${gatewayAddress}`;
+}
+
+// Returns a string that satisfies the format of a transaction hash.
+export function getGenesisFakeTxHash(entity: 'app' | 'supplier' | 'gateway' | 'service', index: number): string {
+  const num = index + 1;
+  let entityId: string
+
+  switch (entity) {
+    case 'app':
+      entityId = 'A';
+      break
+    case 'supplier':
+      entityId = 'B';
+      break
+    case "gateway":
+      entityId = 'C';
+      break
+    case 'service':
+      entityId = 'D';
```

Comment:
Is there a reason not to use the same string as the case (instead of arbitrary letters)? I'm imagining trying to debug this at some point; I think it would make it easier.

Reply:
We use this letters because they are HEX so the hash is still a valid one (string with only HEX and 64 chars)